ShaiRandom/Generators/AbstractRandom.cs

```diff
@@ -38,14 +38,7 @@ protected static ulong MakeSeed()
         protected AbstractRandom()
```

Comment:
Does this have docs now? The old docs wouldn't be accurate anyway.

Reply:
Hm... Actually this brings up a good point; The 0-argument constructor really has nothing to do, since it leaves initialization to the subclasses; in fact, the `protected AbstractRandom(AbstractRandom other)` constructor also isn't used, and could run into the same virtual method issues since it indirectly calls `SetSelectedState` (which is virtual).

The constructor which takes an `AbstractRandom` is actually not used anywhere; and if it is removed the 0-argument constructor is no longer needed (the compiler will generate a default constructor with 0 arguments if there are no custom constructors).  Perhaps we should just remove both constructors?

Reply:
Done in most recent commits.